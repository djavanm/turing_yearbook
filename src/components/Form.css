.student-input {
height: 32px;
border: 1px solid black;
margin: 5px;
font-size: 16px;
text-align: center;
}

.student-button {
  height: 32px;
  border: none;
  margin: auto;
  font-size: 22px;
  font-weight: 600;
  background: none;
  color: white;
  margin-top: 4px;
  display: block;
}

.visible {
  visibility: hidden;
}

#student-button:hover .visible{
  visibility: visible;
}