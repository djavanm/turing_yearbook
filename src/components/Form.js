import React, { Component } from 'react'
import './Form.css'

export default class Form extends Component{
  constructor(props) {
    super(props)
    this.state = {
      name: ''
    }
  }
  updateText = (e) => {
    this.setState({
      name: e.target.value
    })
    console.log(this.props.addStudent)
  }
  submitStudent = (e) =>  {
    e.preventDefault();
    this.props.addStudent(this.state.name);
    this.setState({name: ''});
  }
  render() {
    return (
      <form>
        <input 
          type='text'
          className='student-input'
          placeholder='Student Name'
          name='studentName'
          onChange={e => this.updateText(e)}
          value={this.state.name}
          />
          <button
            className='student-button'
            onClick={e => this.submitStudent(e)}
            >
            Add Student
          </button>
      </form>
    )
  }
}