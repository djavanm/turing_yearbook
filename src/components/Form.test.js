import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  it('matches the Form snapshot created for Form ', () => {
    const addPersonMock = jest.fn();
    const wrapper = shallow(<Form 
        addPerson={addPersonMock}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('calls the add person method on App, on click', () => {
    const addStudentMock = jest.fn();
    const submitStudentMock = jest.fn()
    const wrapper = shallow(<Form 
        addStudent={addStudentMock}/>);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'submitStudent')
    wrapper.find('button').simulate('click', {
      preventDefault: jest.fn()
    });
    expect(instance.submitStudent).toHaveBeenCalled()
  });
});