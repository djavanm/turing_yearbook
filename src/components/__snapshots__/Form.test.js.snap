// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches the Form snapshot created for Form  1`] = `
<form>
  <input
    className="student-input"
    name="studentName"
    onChange={[Function]}
    placeholder="Student Name"
    type="text"
    value=""
  />
  <button
    className="student-button"
    id="student-button"
    onClick={[Function]}
  >
    <span
      className="visible"
    >
      {  
    </span>
    Add Student
    <span
      className="visible"
    >
        }
    </span>
  </button>
</form>
`;
